llm_demo/orchestrator/langchain_tools/langchain_tools_orchestrator.py

```diff
@@ -160,19 +171,37 @@ async def user_session_invoke(self, uuid: str, prompt: str) -> dict[str, Any]:
         response["output"] = agent_response.get("output")
         if confirmation:
             response["confirmation"] = confirmation
+        response["trace"] = self.flush_user_trace(uuid)
         return response
 
-    def user_session_reset(self, session: dict[str, Any], uuid: str):
+    async def user_session_reset(self, session: dict[str, Any], uuid: str):
         user_session = self.get_user_session(uuid)
+
+        # clear session history
         del session["history"]
         base_history = self.get_base_history(session)
         session["history"] = [base_history]
         history = self.parse_messages(session["history"])
         user_session.reset_memory(history)
 
+        # clear session traces
+        del self._user_traces[uuid]
+        tool_trace = ToolTrace()
```

Comment:
do we need to hang on to ToolTrace if it's used by tools?

Reply:
ToolTrace is connected to each user session (to prevent it from sending the wrong tooltrace to another user. (it is similar to _user_sessions. There is another way to do this by initializing it and hanging on to it within UserAgent. What do you think about that?